src/onboarding/steps.test.ts

```diff
@@ -0,0 +1,237 @@
+import { BIOMETRY_TYPE } from 'react-native-keychain'
+import { initializeAccount } from 'src/account/actions'
+import { setHasSeenVerificationNux } from 'src/identity/actions'
+import {
+  navigate,
+  navigateClearingStack,
+  navigateHome,
+  popToScreen,
+} from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import {
+  END_OF_ONBOARDING_SCREEN,
+  FIRST_ONBOARDING_SCREEN,
+  getOnboardingStepValues,
+  goToNextOnboardingScreen,
+  _getScreenInfo,
+} from 'src/onboarding/steps'
+import { store } from 'src/redux/store'
+import { mockOnboardingProps } from 'test/values'
+import { mocked } from 'ts-jest/utils'
+
+jest.mock('src/redux/store', () => ({ store: { dispatch: jest.fn() } }))
+
+const mockStore = mocked(store)
+
+describe('onboarding steps', () => {
+  describe('_getScreenInfo and getOnboardingStepValues', () => {
+    it('All paths from FIRST_ONBOARDING_SCREEN lead to END_OF_ONBOARDING_SCREEN and have incrementing steps', () => {
+      const options = {
+        recoveringFromStoreWipe: [true, false],
+        choseToRestoreAccount: [true, false],
+        supportedBiometryType: [null, BIOMETRY_TYPE.FACE_ID],
+        skipVerification: [true, false],
+        numberAlreadyVerifiedCentrally: [true, false],
+      }
+
+      options.recoveringFromStoreWipe.forEach((recoveringFromStoreWipe) => {
+        options.choseToRestoreAccount.forEach((choseToRestoreAccount) => {
+          options.supportedBiometryType.forEach((supportedBiometryType) => {
+            options.skipVerification.forEach((skipVerification) => {
+              options.numberAlreadyVerifiedCentrally.forEach((numberAlreadyVerifiedCentrally) => {
+                // Initiate Test Variables
+                const onboardingProps = {
+                  recoveringFromStoreWipe,
+                  choseToRestoreAccount,
+                  supportedBiometryType,
+                  skipVerification,
+                  numberAlreadyVerifiedCentrally,
+                }
+                let currentScreen: Screens = FIRST_ONBOARDING_SCREEN
+                let stepCounter = 1
+
+                const navigate = (s: Screens) => {
+                  currentScreen = s
+                  stepCounter++
+                }
+                const navigateHome = () => {
+                  currentScreen = END_OF_ONBOARDING_SCREEN
+                }
+
+                const { totalSteps } = getOnboardingStepValues(
```

Comment:
it looks like all your tests work with the same screen parameter-- would be good to get better coverage on that. the highest impact cases to cover are probably the first screen, the last screen, and some screen in the middle.

Reply:
All of those screen parameters are tested below in the `expect(steps.totalSteps).toEqual(totalSteps)`, testing that throughout the flow the totalCounts are the same.